### ISSUE TITLE
renv_deps fails to include packages with strict versions in DESCRIPTION

### ISSUE THREAD

**Issue by vpnagraj** [signaturescience/pracpac#99]
seeing an issue where if a DESCRIPTION file specifies a minimum version for a package, then the package fails to be included in lockfile after running `renv_deps()`

i think i see *why* this is happening, but we should fix it.

basically the `pracpac:::pkg_info` helper pulls in a list of dependencies exactly as specified in DESCRIPTION. when we go to create an `renv` snapshot with those dependencies, then any with explicit version requirements will not be seen as packages to include in snapshot.

see `reprex` below. 

fix could be to strip out version numbers if present with a regex? maybe inside of `pracpac:::pkg_info()` ? or inside `renv_deps()`? 

cc @stephenturner 

``` r
renv::snapshot(project = tempdir(), 
               packages = c("dplyr", "trending (>= 0.1.0)"), 
               prompt = FALSE, 
               update = TRUE,
               reprex = TRUE)
```

<details style="margin-bottom: 10px;">
<summary>
Lockfile
</summary>

    {
      "R": {
        "Version": "4.2.3",
        "Repositories": [
          {
            "Name": "CRAN",
            "URL": "https://cran.rstudio.com"
          }
        ]
      },
      "Packages": {
        "R6": {
          "Package": "R6",
          "Version": "2.4.1",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "292b54f8f4b94669b08f94e5acce6be2",
          "Requirements": []
        },
        "cli": {
          "Package": "cli",
          "Version": "3.6.0",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "3177a5a16c243adc199ba33117bd9657",
          "Requirements": []
        },
        "dplyr": {
          "Package": "dplyr",
          "Version": "1.1.2",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "dea6970ff715ca541c387de363ff405e",
          "Requirements": [
            "R6",
            "cli",
            "generics",
            "glue",
            "lifecycle",
            "magrittr",
            "pillar",
            "rlang",
            "tibble",
            "tidyselect",
            "vctrs"
          ]
        },
        "fansi": {
          "Package": "fansi",
          "Version": "0.4.1",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "7fce217eaaf8016e72065e85c73027b5",
          "Requirements": []
        },
        "generics": {
          "Package": "generics",
          "Version": "0.1.3",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "15e9634c0fcd294799e9b2e929ed1b86",
          "Requirements": []
        },
        "glue": {
          "Package": "glue",
          "Version": "1.6.2",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "4f2596dfb05dac67b9dc558e5c6fba2e",
          "Requirements": []
        },
        "lifecycle": {
          "Package": "lifecycle",
          "Version": "1.0.3",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "001cecbeac1cff9301bdc3775ee46a86",
          "Requirements": [
            "cli",
            "glue",
            "rlang"
          ]
        },
        "magrittr": {
          "Package": "magrittr",
          "Version": "2.0.3",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "7ce2733a9826b3aeb1775d56fd305472",
          "Requirements": []
        },
        "pillar": {
          "Package": "pillar",
          "Version": "1.9.0",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "15da5a8412f317beeee6175fbc76f4bb",
          "Requirements": [
            "cli",
            "fansi",
            "glue",
            "lifecycle",
            "rlang",
            "utf8",
            "vctrs"
          ]
        },
        "pkgconfig": {
          "Package": "pkgconfig",
          "Version": "2.0.3",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "01f28d4278f15c76cddbea05899c5d6f",
          "Requirements": []
        },
        "rlang": {
          "Package": "rlang",
          "Version": "1.1.1",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "a85c767b55f0bf9b7ad16c6d7baee5bb",
          "Requirements": []
        },
        "tibble": {
          "Package": "tibble",
          "Version": "3.2.1",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "a84e2cc86d07289b3b6f5069df7a004c",
          "Requirements": [
            "fansi",
            "lifecycle",
            "magrittr",
            "pillar",
            "pkgconfig",
            "rlang",
            "vctrs"
          ]
        },
        "tidyselect": {
          "Package": "tidyselect",
          "Version": "1.2.0",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "79540e5fcd9e0435af547d885f184fd5",
          "Requirements": [
            "cli",
            "glue",
            "lifecycle",
            "rlang",
            "vctrs",
            "withr"
          ]
        },
        "utf8": {
          "Package": "utf8",
          "Version": "1.1.4",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "4a5081acfb7b81a572e4384a7aaf2af1",
          "Requirements": []
        },
        "vctrs": {
          "Package": "vctrs",
          "Version": "0.6.2",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "a745bda7aff4734c17294bb41d4e4607",
          "Requirements": [
            "cli",
            "glue",
            "lifecycle",
            "rlang"
          ]
        },
        "withr": {
          "Package": "withr",
          "Version": "2.5.0",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "c0e49a9760983e81e55cdd9be92e7182",
          "Requirements": []
        }
      }
    }

<sup>Lockfile generated by renv 0.16.0.</sup>
</details>

<sup>Created on 2023-05-29 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

``` r
renv::snapshot(project = tempdir(), 
               packages = c("dplyr", "trending"), 
               prompt = FALSE, 
               update = TRUE,
               reprex = TRUE)
```

<details style="margin-bottom: 10px;">
<summary>
Lockfile
</summary>

    {
      "R": {
        "Version": "4.2.3",
        "Repositories": [
          {
            "Name": "CRAN",
            "URL": "https://cran.rstudio.com"
          }
        ]
      },
      "Packages": {
        "MASS": {
          "Package": "MASS",
          "Version": "7.3-51.6",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "1dad32ac9dbd8057167b2979fb932ff7",
          "Requirements": []
        },
        "Matrix": {
          "Package": "Matrix",
          "Version": "1.5-3",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "4006dffe49958d2dd591c17e61e60591",
          "Requirements": [
            "lattice"
          ]
        },
        "R6": {
          "Package": "R6",
          "Version": "2.4.1",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "292b54f8f4b94669b08f94e5acce6be2",
          "Requirements": []
        },
        "Rcpp": {
          "Package": "Rcpp",
          "Version": "1.0.10",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "e749cae40fa9ef469b6050959517453c",
          "Requirements": []
        },
        "RcppEigen": {
          "Package": "RcppEigen",
          "Version": "0.3.3.9.3",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "1e035db628cefb315c571202d70202fe",
          "Requirements": [
            "Matrix",
            "Rcpp"
          ]
        },
        "abind": {
          "Package": "abind",
          "Version": "1.4-5",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "4f57884290cc75ab22f4af9e9d4ca862",
          "Requirements": []
        },
        "arm": {
          "Package": "arm",
          "Version": "1.13-1",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "ecd1fd4d4ea49df1b62fb706c37785ac",
          "Requirements": [
            "MASS",
            "Matrix",
            "abind",
            "coda",
            "lme4",
            "nlme"
          ]
        },
        "boot": {
          "Package": "boot",
          "Version": "1.3-25",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "bd51734a754b6c2baf28b2d1ebc11e91",
          "Requirements": []
        },
        "ciTools": {
          "Package": "ciTools",
          "Version": "0.6.1",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "b4acc5f619ee2f36d6611a0a3ad5c40e",
          "Requirements": [
            "MASS",
            "arm",
            "boot",
            "dplyr",
            "lme4",
            "survival"
          ]
        },
        "cli": {
          "Package": "cli",
          "Version": "3.6.0",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "3177a5a16c243adc199ba33117bd9657",
          "Requirements": []
        },
        "coda": {
          "Package": "coda",
          "Version": "0.19-4",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "24b6d006b8b2343876cf230687546932",
          "Requirements": [
            "lattice"
          ]
        },
        "dplyr": {
          "Package": "dplyr",
          "Version": "1.1.2",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "dea6970ff715ca541c387de363ff405e",
          "Requirements": [
            "R6",
            "cli",
            "generics",
            "glue",
            "lifecycle",
            "magrittr",
            "pillar",
            "rlang",
            "tibble",
            "tidyselect",
            "vctrs"
          ]
        },
        "fansi": {
          "Package": "fansi",
          "Version": "0.4.1",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "7fce217eaaf8016e72065e85c73027b5",
          "Requirements": []
        },
        "generics": {
          "Package": "generics",
          "Version": "0.1.3",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "15e9634c0fcd294799e9b2e929ed1b86",
          "Requirements": []
        },
        "glue": {
          "Package": "glue",
          "Version": "1.6.2",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "4f2596dfb05dac67b9dc558e5c6fba2e",
          "Requirements": []
        },
        "lattice": {
          "Package": "lattice",
          "Version": "0.20-45",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "b64cdbb2b340437c4ee047a1f4c4377b",
          "Requirements": []
        },
        "lifecycle": {
          "Package": "lifecycle",
          "Version": "1.0.3",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "001cecbeac1cff9301bdc3775ee46a86",
          "Requirements": [
            "cli",
            "glue",
            "rlang"
          ]
        },
        "lme4": {
          "Package": "lme4",
          "Version": "1.1-33",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "2fa7b59feca0d767533fd0c8a7a2bf4b",
          "Requirements": [
            "MASS",
            "Matrix",
            "Rcpp",
            "RcppEigen",
            "boot",
            "lattice",
            "minqa",
            "nlme",
            "nloptr"
          ]
        },
        "magrittr": {
          "Package": "magrittr",
          "Version": "2.0.3",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "7ce2733a9826b3aeb1775d56fd305472",
          "Requirements": []
        },
        "minqa": {
          "Package": "minqa",
          "Version": "1.2.5",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "587ce77fd3c7bada7eadb2d18b62930d",
          "Requirements": [
            "Rcpp"
          ]
        },
        "nlme": {
          "Package": "nlme",
          "Version": "3.1-162",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "0984ce8da8da9ead8643c5cbbb60f83e",
          "Requirements": [
            "lattice"
          ]
        },
        "nloptr": {
          "Package": "nloptr",
          "Version": "1.2.2.2",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "2737faeee353704efec5afa1e943dd64",
          "Requirements": []
        },
        "pillar": {
          "Package": "pillar",
          "Version": "1.9.0",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "15da5a8412f317beeee6175fbc76f4bb",
          "Requirements": [
            "cli",
            "fansi",
            "glue",
            "lifecycle",
            "rlang",
            "utf8",
            "vctrs"
          ]
        },
        "pkgconfig": {
          "Package": "pkgconfig",
          "Version": "2.0.3",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "01f28d4278f15c76cddbea05899c5d6f",
          "Requirements": []
        },
        "rlang": {
          "Package": "rlang",
          "Version": "1.1.1",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "a85c767b55f0bf9b7ad16c6d7baee5bb",
          "Requirements": []
        },
        "survival": {
          "Package": "survival",
          "Version": "3.5-3",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "aea2b8787db7088ba50ba389848569ee",
          "Requirements": [
            "Matrix"
          ]
        },
        "tibble": {
          "Package": "tibble",
          "Version": "3.2.1",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "a84e2cc86d07289b3b6f5069df7a004c",
          "Requirements": [
            "fansi",
            "lifecycle",
            "magrittr",
            "pillar",
            "pkgconfig",
            "rlang",
            "vctrs"
          ]
        },
        "tidyselect": {
          "Package": "tidyselect",
          "Version": "1.2.0",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "79540e5fcd9e0435af547d885f184fd5",
          "Requirements": [
            "cli",
            "glue",
            "lifecycle",
            "rlang",
            "vctrs",
            "withr"
          ]
        },
        "trending": {
          "Package": "trending",
          "Version": "0.1.0",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "9559633c7beb1c8a9323785818af3d2a",
          "Requirements": [
            "MASS",
            "ciTools",
            "pillar",
            "tibble",
            "vctrs"
          ]
        },
        "utf8": {
          "Package": "utf8",
          "Version": "1.1.4",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "4a5081acfb7b81a572e4384a7aaf2af1",
          "Requirements": []
        },
        "vctrs": {
          "Package": "vctrs",
          "Version": "0.6.2",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "a745bda7aff4734c17294bb41d4e4607",
          "Requirements": [
            "cli",
            "glue",
            "lifecycle",
            "rlang"
          ]
        },
        "withr": {
          "Package": "withr",
          "Version": "2.5.0",
          "Source": "Repository",
          "Repository": "CRAN",
          "Hash": "c0e49a9760983e81e55cdd9be92e7182",
          "Requirements": []
        }
      }
    }

<sup>Lockfile generated by renv 0.16.0.</sup>
</details>

<sup>Created on 2023-05-29 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by stephenturner** [comment]
I think this should do it 2ef06e2

But it's a blunt fix. If there's a minimum version specified in the description of the package you're pracpac'ing, it'll just strip it out completely. But, this should be okay I think. If the package you're pracpac'ing has a minimum version, presumably that minimum version is installed on the machine you're developing on, because the package must be for renv to work at all. 

This doesn't deal with _exact_ version specifications, but [this is strongly recommended against anyway](https://r-pkgs.org/description.html#sec-description-imports-suggests-minium-version).

**Comment by stephenturner** [comment]
One little update in c9177fd. Turns out `pkgname(>= 0.1.0)` (i.e., without whitespace) is fine. c9177fd captures whitespace, parens, etc.

**Comment by vpnagraj** [comment]
thanks @stephenturner ! 

should be completely fixed in #100 

noting that the way the `imports` object came through we actually needed to *split* before the `gsub()`.


``` r
imports <- "dplyr,\nfable,\nfabletools (>= 0.3.2),\nlubridate,\nmagrittr,\nMMWRweek,\npurrr,\ntibble,\ntidyr,\ntsibble,\ntrending (>= 0.1.0),\ntrendeval (>= 0.1.0),\nggplot2,\nstringr,\njsonlite,\nstats,\nscales,\nreadr,\nshiny,\nhttr"

imports <- gsub("[ \\(<=>].*", "", imports)

imports
#> [1] "dplyr,\nfable,\nfabletools"
```

<sup>Created on 2023-05-30 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

``` r
imports <- "dplyr,\nfable,\nfabletools (>= 0.3.2),\nlubridate,\nmagrittr,\nMMWRweek,\npurrr,\ntibble,\ntidyr,\ntsibble,\ntrending (>= 0.1.0),\ntrendeval (>= 0.1.0),\nggplot2,\nstringr,\njsonlite,\nstats,\nscales,\nreadr,\nshiny,\nhttr"

imports <- strsplit(imports, split=",\\n")[[1]]
# Strip out any version requirements
imports <- sapply(imports, function(x) gsub("[ \\(<=>].*", "", x), USE.NAMES = FALSE)

imports
#>  [1] "dplyr"      "fable"      "fabletools" "lubridate"  "magrittr"  
#>  [6] "MMWRweek"   "purrr"      "tibble"     "tidyr"      "tsibble"   
#> [11] "trending"   "trendeval"  "ggplot2"    "stringr"    "jsonlite"  
#> [16] "stats"      "scales"     "readr"      "shiny"      "httr"
```

<sup>Created on 2023-05-30 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
